src/utils/form/abstract-control.ts

```diff
@@ -43,6 +44,8 @@ export abstract class AbstractControl {
     public abstract get readonly(): boolean;
     public abstract set readonly(isReadonly: boolean);
     public abstract get touched(): boolean;
+    public abstract get controls(): AbstractControl[];
+    public abstract getControl<T = any>(name: string): AbstractControl<T>;
```

Comment:
est-ce que tous les enfants pourront répondre à ces nouvelles méthodes (getControl et le set/set value), parce que si c'est quelque chose qui n'est pas faisable dans certaines classes enfant, je pense que c'est une mauvaise pratique OO de le mettre dans une classe de base et obliger les enfants qui n'ont pas la capacité de répondre à ce contrat d'en hériter, il faudrait sortir des interfaces avec ces méthodes peut-être? des throw unsupported c'est pas super facile à déceler lorsque tu utilises une classe et call une méthode existante pour laquelle tu attends un résultat..

Reply:
J'ai implémenté (suite au code review) aux endroits faciles, mais aux endroits plus difficile ça ira à plus tard (mais que ça soit possible un jour). En effet, il faut faire en sorte d'éviter qu'il y ait un endroit où c'est impossible.

Reply:
Je suis d'accord avec @jsroy19  , l'implémentation de abstract-control devrait contenir seulement des propriété et méthode qui sont partagé par tous les enfant , c'est a dire FormGroup , FormArray et FonrmControl 

je pense que si on ajoute des trucs qui sont relié a FormGroup et FormArray on dénature un peu l'objet, un control (AbstractControl) ne devrait pas avoir une notion de Groupe de control, du moins c'était l'idée original,

Reply:
IMO si c'est pour améliorer la lisibilité et l'utilisabilité c'est pas si pire que FormControl retourne undefined sur le getControl.

Je suis all in sur les beaux concepts d'orientés objets, mais dans ce cas là
1- Il y a quand même une implémentation logique.  Un peu plus tordue, mais logique.  Le FormControl peut toujours retourner undefined (un lookup dans les contrôles qu'ils possèdent (aucun) ne peut jamais retourner de résultat).
2- Le fait de permettre cette implémentation simplifie le code qu'on a à faire.  Pu de cast, on passe tous par le même interface sans savoir on deal avec quel genre de contrôle.  À l'usage, ce qui nous intéresse souvent c'est la valeur et les erreurs.  Le reste, que ce soit un FormControl, un FormArray ou un FormGroup, ça nous intéresse très peu.

FormGroup et Form Array sont des aggrégats de contrôles.  Je ne vois pas pourquoi FormControl pourrait pas en être un lui aussi fonctionnelement.  Qu'il ne puisse posséder qu'un contrôle, c'est un détail d'implémentation.

Reply:
je suis en paix finalement avec ça.